pkg/routeagent_driver/handlers/ovn/connection.go

```diff
@@ -0,0 +1,195 @@
+		if err != nil {
+			return errors.Wrapf(err, "error getting config for %q", getOVNPrivKeyPath())
+		}
+
+		caFile, err := clusterfiles.Get(c.k8sClientset, getOVNCaBundlePath())
+		if err != nil {
+			return errors.Wrapf(err, "error getting config for %q", getOVNCaBundlePath())
+		}
+
+		tlsConfig, err = getOVNTLSConfig(pkFile, certFile, caFile)
+		if err != nil {
+			return errors.Wrap(err, "error getting OVN TLS config")
+		}
+	}
+
+	// Create nbdb client
+	nbdbModel, err := nbdb.FullDatabaseModel()
+	if err != nil {
+		return errors.Wrap(err, "error getting OVN NBDB database model")
+	}
+
+	c.nbdb, err = c.createLibovsdbClient(getOVNNBDBAddress(), tlsConfig, nbdbModel)
+	if err != nil {
+		return errors.Wrap(err, "error creating NBDB connection")
+	}
+
+	return nil
+}
+
+func getOVNTLSConfig(pkFile, certFile, caFile string) (*tls.Config, error) {
+	cert, err := tls.LoadX509KeyPair(certFile, pkFile)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failure loading ovn certificates")
+	}
+
+	rootCAs := x509.NewCertPool()
+
+	data, err := os.ReadFile(caFile)
+	if err != nil {
+		return nil, errors.Wrap(err, "failure loading OVNDB ca bundle")
+	}
+
+	rootCAs.AppendCertsFromPEM(data)
+
+	return &tls.Config{
+		Certificates: []tls.Certificate{cert},
+		RootCAs:      rootCAs,
+		ServerName:   "ovn",
+		MinVersion:   tls.VersionTLS12,
+	}, nil
+}
+
+func (c *ConnectionHandler) createLibovsdbClient(dbAddress string, tlsConfig *tls.Config,
+	dbModel model.ClientDBModel,
+) (libovsdbclient.Client, error) {
+	options := []libovsdbclient.Option{
+		// Reading and parsing the DB after reconnect at scale can (unsurprisingly)
+		// take longer than a normal ovsdb operation. Give it a bit more time so
+		// we don't time out and enter a reconnect loop.
+		libovsdbclient.WithReconnect(OVSDBTimeout, &backoff.ZeroBackOff{}),
+		libovsdbclient.WithLogger(&logger.Logger),
+	}
+
+	if strings.HasPrefix(dbAddress, "IC:") {
+		node, err := nodeutil.GetLocalNode(c.k8sClientset)
+		if err != nil {
+			return nil, errors.Wrap(err, "error getting the node")
+		}
+
+		annotations := node.GetAnnotations()
+
+		zoneName, ok := annotations[constants.OvnZoneAnnotation]
+		if !ok {
+			return nil, errors.Wrapf(err, "node %q is missing the %q "+
+				"annotation", node.Name, constants.OvnZoneAnnotation)
+		}
+
+		zoneDBAddress := getICDBAddress(dbAddress, zoneName)
+		options = append(options, libovsdbclient.WithEndpoint(zoneDBAddress))
+	} else {
+		options = append(options, libovsdbclient.WithEndpoint(dbAddress))
+	}
+
+	if tlsConfig != nil {
+		options = append(options, libovsdbclient.WithTLSConfig(tlsConfig))
+	}
+
+	client, err := libovsdbclient.NewOVSDBClient(dbModel, options...)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating ovsdbClient")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), OVSDBTimeout)
+	defer cancel()
+
+	err = client.Connect(ctx)
+
+	err = errors.Wrap(err, "error connecting to ovsdb")
```

Comment:
`Wrap` handles that.

Reply:
Okay thanks. But since we have a check for `err != nil` at Line176, it would be good to move it there as it improves readability.

My 2 cents: Normally I prefer code that is readable in the first attempt, rather than expecting someone to know that Wrap handles it internally ;-)